clang/Type.d

```diff
@@ -93,31 +110,53 @@ struct Type
         return isObjCIdType || isObjCClassType || isObjCSelType;
     }
 
+    @property bool isPointer ()
+    {
+        return kind == CXTypeKind.CXType_Pointer;
+    }
+
+    @property bool isTypedef ()
+    {
+        return kind == CXTypeKind.CXType_Typedef;
+    }
+
+    @property bool isValid ()
+    {
+        return kind != CXTypeKind.CXType_Invalid;
+    }
+
     @property bool isWideCharType ()
     {
-        with (CXTypeKind)
-            return kind == CXType_WChar;
+        return kind == CXTypeKind.CXType_WChar;
     }
 
-    @property bool isConst ()
+    @property Type canonical()
     {
-        return clang_isConstQualifiedType(cx) == 1;
+        return canonical_ ? *canonical_ : Type(clang_getCanonicalType(cx));
     }
 
-    @property bool isExposed ()
+    @property void canonical(Type type)
```

Comment:
So far a type has been immutable, in the sense that it did not have any setters.